File: Recien traducido/Matematica/IsMagicSquare.java
// JAVA program to check whether a given 
// matrix is magic matrix or not 
  
import java.io.*; 
  
class GFG { 
      
    static int N = 3; 
      
    // Returns true if mat[][] is magic 
    // square, else returns false. 
    static boolean isMagicSquare(int mat[][]) 
    { 
          
        // calculate the sum of 
        // the prime diagonal 
         int sum = 0,sum2=0;  
        for (int i = 0; i < N; i++) 
            sum = sum + mat[i][i]; 
  
        // the secondary diagonal 
        for (int i = 0; i < N; i++) 
            sum2 = sum2 + mat[i][N-1-i]; 
  
        if(sum!=sum2)  
            return false; 
  
        // For sums of Rows 
        for (int i = 0; i < N; i++) { 
  
            int rowSum = 0; 
            for (int j = 0; j < N; j++) 
                rowSum += mat[i][j]; 
  
            // check if every row sum is 
            // equal to prime diagonal sum 
            if (rowSum != sum) 
                return false; 
        } 
  
        // For sums of Columns 
        for (int i = 0; i < N; i++) { 
  
            int colSum = 0; 
            for (int j = 0; j < N; j++) 
                colSum += mat[j][i]; 
  
            // check if every column sum is 
            // equal to prime diagonal sum 
            if (sum != colSum) 
                return false; 
        } 
  
        return true; 
    } 
  
    // driver program to 
    // test above function 
    public static void main(String[] args) 
    { 
        int mat[][] = {{ 2, 7, 6 }, 
                       { 9, 5, 1 }, 
                       { 4, 3, 8 }}; 
  
        if (isMagicSquare(mat)) 
            System.out.println("Magic Square"); 
        else
            System.out.println("Not a magic" + 
                                    " Square"); 
    } 
} 

File: Recien traducido/Matematica/IsMagicSquare.py

N = 3

def isMagicSquare(mat): 
    s = 0 
    for i in range(0, N): 
        s = s + mat[i][i] 

    s2 = 0
    for i in range(0, N): 
        s2 = s2 + mat[i][N-i-1] 
  
    if(s != s2): 
        return False
 
    for i in range(0, N): 
        rowSum = 0;      
        for j in range(0, N): 
            rowSum += mat[i][j] 
         
        if (rowSum != s): 
            return False

    for i in range(0, N): 
        colSum = 0
        for j in range(0, N): 
            colSum += mat[j][i] 
        if (s != colSum): 
            return False
  
    return True
   
mat = [[2, 7, 6], 
    [9, 5, 1], 
    [4, 3, 8]] 
      
if (isMagicSquare(mat)): 
    print("Magic Square") 
else: 
    print("Not a magic Square") 